### PR TITLE
Use travis stages

### PR DESCRIPTION
**Pull request by raboof**
refs #27015

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -26,19 +27,23 @@ cache:
...
-  - sbt -jvm-opts .jvmopts-travis -Dakka.build.scalaVersion=$TRAVIS_SCALA_VERSION ";update ;mimaReportBinaryIssues ;test:compile ;validateCompile"
-  # make 'git branch' work again
-  - git branch -f "$TRAVIS_BRANCH" && git checkout "$TRAVIS_BRANCH"
-  # check policies, if on master also upload
-  - if [[ "$TRAVIS_SECURE_ENV_VARS" == "true" ]]; then if [[ "$TRAVIS_BRANCH" == "master" && "$TRAVIS_EVENT_TYPE" == "push" ]]; then sbt 'set credentials += Credentials("whitesource", "whitesourcesoftware.com", "", System.getenv("WHITESOURCE_KEY"))' whitesourceCheckPolicies whitesourceUpdate; else sbt 'set credentials += Credentials("whitesource", "whitesourcesoftware.com", "", System.getenv("WHITESOURCE_KEY"))' whitesourceCheckPolicies; fi ; fi
+jobs:
+  - stage: whitesource
+    script: git branch -f "$TRAVIS_BRANCH" && git checkout "$TRAVIS_BRANCH" && sbt whitesourceCheckPolicies whitesourceUpdate
+  - stage: compile_and_mima
+    script: sbt -jvm-opts .jvmopts-travis -Dakka.build.scalaVersion=$TRAVIS_SCALA_VERSION ";update ;mimaReportBinaryIssues ;test:compile ;validateCompile"
```
Comment (2m) [inline .travis.yml]: There is also a `name` job property for more human readable job names.

**Review by 2m — APPROVED** [review]
LGTM

**Comment by raboof** [comment]
failure was intermittent ssh problem:
```
+ rm -rf /localhome/jenkinsakka/pr-validator-per-commit-jenkins/16132
+ ssh a5.local rm -rf /localhome/jenkinsakka/pr-validator-per-commit-jenkins/16132
+ ssh a6.local rm -rf /localhome/jenkinsakka/pr-validator-per-commit-jenkins/16132
Permission denied, please try again.
Permission denied, please try again.
Permission denied (publickey,password).
```

**Comment by raboof** [comment]
PLS BUILD

File: .travis.yml [hunk 2]
```diff
@@ -9,7 +9,8 @@ scala:
 before_install:
   # using jabba for custom jdk management
   - curl -sL https://raw.githubusercontent.com/shyiko/jabba/0.11.2/install.sh | bash && . ~/.jabba/jabba.sh
-  - jabba install adopt@~1.8.202-08
+  - jabba install $TRAVIS_JDK
+install: jabba use "$TRAVIS_JDK" && java -Xmx32m -version
```
Comment (helena) [inline .travis.yml]: 👍

**Review by helena — APPROVED** [review]
Nice!

**Comment by helena** [comment]
@raboof I kicked off a re-build which should fix the perms issue on the Jenkins PR validation.

**Pull request merged**